autocorrect/src/rule/halfwidth.rs

```diff
@@ -133,28 +129,29 @@ pub fn format_punctuation(text: &str) -> String {
 }
 
 pub fn format_word(text: &str) -> String {
-    let mut out = String::new();
-
-    for part in text.split("") {
-        if let Some(new_str) = CHAR_WIDTH_MAP.get(part) {
-            out.push_str(new_str);
-            continue;
-        }
-
-        out.push_str(part);
-    }
+    let out = text
+        .chars()
+        .map(|c| match c {
+            '\u{FF10}'..='\u{FF19}' | '\u{FF21}'..='\u{FF3A}' | '\u{FF41}'..='\u{FF5A}' => {
+                // checked char is in range of fullwidth number and alphabetic
+                unsafe { char::from_u32_unchecked(c as u32 - 0xFEE0) }
+            }
+            '\u{3000}' => ' ',
```

Comment:
你怎么知道我之前那个看不见的空格是 `\u{3000}` 的? 😂

Reply:
https://en.wikipedia.org/wiki/Halfwidth_and_Fullwidth_Forms_(Unicode_block)

> U+FF00 does not correspond to a fullwidth ASCII 20 (space character), since that role is already fulfilled by U+3000 "[ideographic space](https://en.wikipedia.org/wiki/Ideographic_space)".